### PR TITLE
[bisect] add rel_error argument

### PR DESCRIPTION
**Pull request by tobiasleibner**

### CODE REVIEW

**Comment by renefritze** [comment]
Rebase on master should fix CI.

File: dune/xt/common/bisect.hh [hunk 1]
```diff
@@ -22,27 +22,31 @@ namespace Common {
...
+ *        any y > x.
  *
  * \note Presumes that: if condition(x_1) == condition(x_2) == value,
  *                      then there does not exist a x_1 < y < x_2, s.t. condition(y) == !value;
  */
-double find_largest_by_bisection(const double& left, const double& right, std::function<bool(const double&)> condition)
+double find_largest_by_bisection(const double& left,
+                                 const double& right,
+                                 std::function<bool(const double&)> condition,
+                                 const double rel_error = 1e-16,
```
Comment (renefritze) [inline dune/xt/common/bisect.hh]: Maybe get those defaults from the FloatCmp defaults for double? That would be more consistent.

Comment (tobiasleibner) [reply dune/xt/common/bisect.hh]: Done. Also had to change the test, as ``EXPECT_DOUBLE_EQ`` seems to use a smaller default tolerance.

Comment (renefritze) [reply dune/xt/common/bisect.hh]: There are shortcuts for that: `DXTC_EXPECT_FLOAT_EQ` and so forth from https://github.com/dune-community/dune-xt-common/blob/master/dune/xt/common/test/float_cmp_generated.hxx (which should not be included directly, but rather https://github.com/dune-community/dune-xt-common/blob/master/dune/xt/common/test/float_cmp.hh)

Comment (tobiasleibner) [reply dune/xt/common/bisect.hh]: Okay, used these shortcuts now.

**Pull request merged**